### DISCUSSION TITLE
Add comparison with some other periods

### DISCUSSION THREAD

**Discussion by divinerites** [plausible/analytics#277] [Anything Else]
## Prerequisites
- [X] I have searched open and closed issues to make sure that the feature has not yet been requested.

## Feature request

Being able to compare the actual showed period with an other would be a good & useful feature.

There is 2 main use case :

1. Same period last year : *(7 days <-> same dates last year)*
1. The period just before this one : *(7 days <-> the 7 days just before)*

There is a minor use case : Use `last month/trimester/semester` instead of `last year` when this option makes sense.

**How would you like it to work**

Add a checkbox/selecter and :

- Shows the curve in an other color
- Shows the drilldowns in 2 columns (top pages, top sources, countries, goals)

The perfect UI would let us choose whether we see : actual period, old period, both periods.

**Update:** You can now compare your stats over time! Select "**Compare**" in the date picker or press the "**X**" key on your keyboard to start comparing. Learn more about the comparisons here: https://plausible.io/docs/compare-stats.

**Comment by divinerites** [comment]
This also works for https://github.com/plausible/analytics/issues/126

**Comment by Lucent** [comment]
One big improvement over Google Analytics would be if "compared to one year ago" shifted to the nearest same weekday.

**Comment by metmarkosaric** [comment]
Some more ideas for extra comparisons here: https://github.com/plausible/analytics/discussions/391#discussioncomment-216436

"I feel like you should be able to select the comparison period (which is how e.g. Stripe does it) as opposed to inferring it. E.g., how can I compare November to January? I realise that's another menu but would take out some of the ambiguity. Re. the above, I'm personally more interesting in the direct comparison to the previous month (Nov 1–16 vs Dec 1–16), but appreciate the other one might be more popular."

**Comment by shornuk** [comment]
Just to add to this. New customer on trial and liking it so far. Comparison is one of the big things I currently use GA for. I frequently compare any set custom period with the same period the previous year to see how things are progressing.

**Reply by metmarkosaric** [reply]
thanks for your feedback @shornuk! year over year comparison for our custom range would be very useful!

**Comment by shornuk** [comment]
Something like this would be great...
![comparison-dropdown](https://user-images.githubusercontent.com/2061819/105381211-abfd5e80-5c06-11eb-9aea-17553361b666.png)

Especially if it tied in with the comparison lines on the chart as per https://github.com/plausible/analytics/discussions/532

**Reply by metmarkosaric** [reply]
That's exactly what we plan to do! We'll just make the comparison menu fit into the current date picker so it takes less space. And when you view a comparison you also get the comparison line in the chart. @Vigasaurus has some ideas and will help us make this happen!

**Reply by shornuk** [reply]
Awesome. I'm halfway through my trial at the moment, but I can see myself switching off GA in a couple of weeks! 👍

**Reply by ingejanse** [reply]
What is the status of this one? I'm really looking forward to this function, also being halfway my trial at the moment.

**Reply by ukutaht** [reply]
There are a couple big projects that need to be integrated first (GA import, main graph selections, intervals, there are PRs for each). After those features, comparisons are very high on the list.

**Reply by shornuk** [reply]
Great. If we get GA import at some point that's a major plus. (But I'll be longing even more for comparisons 😂)

**Comment by shornuk** [comment]
Still waiting for this one too! I’ve gone past my first full year, so the need for comparisons is getting greater.

**Reply by System76Coffey** [reply]
One more for this feature! Just comparing to the prior period can't take into account special promotions or other outliers, but YOY comparisons allow you to compare in a much more apples to apples way, especially if your promotion always run during similar times in the calendar.

**Reply by System76Coffey** [reply]
@metmarkosaric Is there an updated ETA for this feature?

**Reply by metmarkosaric** [reply]
No ETA at the moment unfortunately. There are a couple big projects that need to be integrated first (GA import, main graph selections, intervals, there are PRs for each). After those features, comparisons are very high on the list.

**Reply by erskingardner** [reply]
Looking forward to this feature a lot!

**Comment by osckarl** [comment]
Messaged the team today to ask about if there's an updated ETA. There's none at the moment. However, they did mention that after intervals are released (sounded soon), this will be one of the features included in the "next sprint". Hopefully soon!

**Reply by ukutaht** [reply]
Correct. Intervals is in final stages of polishing and unless something urgent comes up internally, we'll start on comparisons soon after that.

**Reply by osckarl** [reply]
Thanks. Is this now planned? Is there anything you can share in terms of ETA?

**Reply by ukutaht** [reply]
It is planned but we also have a couple weeks of other work to do first. We're a small team and we have to juggle between working on infrastructure, operations, accuracy, bugfixing, and also adding new features. Comparisons is the next dashboard feature we're planning to work on but we have a lot of other areas of focus as well, so I'm sorry if things aren't going at the speed you're expecting.

We don't share ETAs publicly because of these reasons

**Reply by osckarl** [reply]
Obviously, no need to apologize for that. I'm sure you are doing all in your power to juggle the maintenance with new features. The new KPIs are great, and the intervals were epic too. I feel though as this is such a strong (and common) product offering in your sphere. Keep us updated and keep up the good work!

**Reply by metmarkosaric** [reply]
hi @osckarl! glad you liked the recent updates! work's in progress on comparisons.you can see how it currently looks like here: https://github.com/plausible/analytics/pull/2753. stay tuned for an update!

**Comment by metmarkosaric** [comment]
**Update:** You can now compare your stats over time! Select "**Compare**" in the date picker or press the "**X**" key on your keyboard to start comparing. Learn more about the comparisons here: https://plausible.io/docs/compare-stats.

**Reply by shornuk** [reply]
Nice one! 👍

**Reply by 2803media** [reply]
Hi thanks for this update, do you have any idea for a release date with this feature?

**Reply by metmarkosaric** [reply]
hi @2803media! it's live already

**Reply by 2803media** [reply]
> hi @2803media! it's live already

Sorry my question was about the self hosted version ;)

**Reply by 2803media** [reply]
I guess I must change the link to the 2.0 in docker conf https://hub.docker.com/r/plausible/analytics/tags is that right?

**Reply by metmarkosaric** [reply]
this is not live in self-hosted. self-hosted is a long term release so new features are not available immediately. you can read more about the differences here https://plausible.io/self-hosted-web-analytics

**Reply by ruslandoga** [reply]
@2803media that tag is not officially released yet. It was built to verify https://github.com/plausible/analytics/pull/2879 works. There still might be some other problems. However, if you are set on trying it out, please see the WIP release notes: https://github.com/plausible/hosting/issues/68

**Reply by ruslandoga** [reply]
@2803media https://github.com/plausible/analytics/discussions/2896 has been (pre-)released :)
